dbt/include/redshift/macros/adapters/unit_testing.sql

```diff
@@ -0,0 +1,11 @@
+{%- macro redshift__validate_fixture_rows(rows, row_number) -%}
+  {%- if rows is not none and rows|length > 0 -%}
+    {%- set row = rows[0] -%}
+    {%- for key, value in row.items() -%}
+      {%- if value is none -%}
+          {%- set fixture_name = "expected output" if model.resource_type == 'unit_test' else ("'" ~ model.name ~ "'") -%}
+          {{ exceptions.raise_compiler_error("Unit test fixture " ~ fixture_name ~ " in " ~ model.name ~ " does not have any row free of null values, which may cause type mismatch errors during unit test execution.") }}
```

Comment:
Did we still want to raise an error here? I remember we talked about potentially just a warning instead of an error, but that could be old information.

Reply:
Okay, so Michelle and I aligned on this, and I'll see what you think!

Basically there is a special case that we're still throwing an error in Redshift (there is no all-`None` value row to promote to the first row of the fixture under the hood). 

This special case throws a cryptic error for redshift users specifically.

So now we throw a helpful error that documents this behavior both for user in logs and us in code.

I proposed that this does not introduce new behavior because this was already an error and will continue to be.

The logic change made in core prevents a large percentage of the time this exception would be thrown.